### PR TITLE
remove java methods

### PR DESCRIPTION
**Pull request by dsvag**
just preparation for multiplatform

### CODE REVIEW

**Review by opengamer29 — APPROVED** [review]

File: elmslie-core/src/main/java/vivid/money/elmslie/core/store/Result.kt [hunk 1]
```diff
@@ -13,8 +13,8 @@ data class Result<State : Any, Effect : Any, Command : Any>(
         command: Command? = null,
     ) : this(
         state = state,
-        effects = effect?.let(::listOf) ?: emptyList(),
```
Comment (rcmkt) [inline elmslie-core/src/main/java/vivid/money/elmslie/core/store/Result.kt:16]: why this is required for KMP? yeah, this line is simplified but it also creates additional list for filtering

File: gradle/libs.versions.toml [hunk 2]
```diff
@@ -3,7 +3,7 @@ agp = "8.0.2"
 coroutines = "1.7.1"
 dokka = "1.8.20"
 junit = "5.9.3"
-kotlin = "1.8.21"
```
Comment (rcmkt) [inline gradle/libs.versions.toml:6]: I think we shouldn't bump this for now

File: elmslie-core/src/main/java/vivid/money/elmslie/core/store/Result.kt (hunk 1 above)
Comment (dsvag) [reply elmslie-core/src/main/java/vivid/money/elmslie/core/store/Result.kt:16]: under the hood it works exactly the same but looks more readble 
`public fun <T : Any> listOfNotNull(element: T?): List<T> = if (element != null) listOf(element) else emptyList()`

File: gradle/libs.versions.toml (hunk 2 above)
Comment (dsvag) [reply gradle/libs.versions.toml:6]: yeah we can do it later

**Review by rcmkt — APPROVED** [review]

**Review by sergei-lapin — APPROVED** [review]

**Pull request merged**